### PR TITLE
Support Koncorde syntax for document:count

### LINKED ISSUE #2381: Support Koncorde syntax for `document:count`
## Feature description

The API action `document:count` does not support [Koncorde filters syntax](https://docs.kuzzle.io/core/2/guides/main-concepts/querying/#koncorde-query) for querying like other methods involving a search query like `document:search` for example.

An example of implementation for [document:deleteByQuery](https://github.com/kuzzleio/kuzzle/blob/master/lib/api/controllers/documentController.js#L776)

### PR DESCRIPTION
**Pull request by OlivierCavadenti**
## What does this PR do ?

Support [Koncorde filters syntax](https://docs.kuzzle.io/core/2/guides/main-concepts/querying/#koncorde-query) for querying like other search methods.

### How should this be manually tested?

Use Koncorde filters in query parameters and set "lang" parameter to "koncorde".

Closes #2381

### CODE REVIEW

File: doc/2/api/controllers/document/count/index.md [hunk 1]
```diff
@@ -65,7 +72,8 @@ kourou document:count <index> <collection> <body>
 
 ### Optional:
 
-- `query`: if provided, will count only documents matching this search query. Uses the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) syntax.
+- `query`: the search query itself, using the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) or the [Koncorde Filters DSL](/core/2/api/koncorde-filters-syntax) syntax.
+- `lang`: specify the query language to use. By default, it's `elasticsearch` but `koncorde` can also be used. <SinceBadge version="2.8.0"/>
```
Comment (Shiranuit) [inline doc/2/api/controllers/document/count/index.md:76]:
```suggestion
- `lang`: specify the query language to use. By default, it's `elasticsearch` but `koncorde` can also be used. <SinceBadge version="auto-version"/>
```

Comment (Shiranuit) [reply doc/2/api/controllers/document/count/index.md:76]: You should use `auto-version` this way the version is replaced  with the proper value when released.

**Review by Shiranuit — CHANGES_REQUESTED** [review]

**Comment by kuzzle** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Found some issues !</td>
    </tr>
  </tbody>
</table>



## Executed Checks

### :heavy_check_mark: Branch Merge Check



### :x: Changelog Tag Check

Pull Requests must have a `changelog:XXX` label.
If you want to allow this PR to not have a changelog tag, add the following line to the PR body:


```
ci-no-changelog-tag
```

### :heavy_check_mark: PR Body Check


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4a04369c04f72dd6f4a3e352d6f3993762c77082
</p>

**Review by Aschen — APPROVED** [review]

**Review by Shiranuit — APPROVED** [review]

**Comment by Aschen** [comment]
You can comment the dead link check job

**Pull request merged**